cliconfig/config.go

```diff
@@ -0,0 +1,215 @@
+package cliconfig
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/docker/docker/pkg/homedir"
+)
+
+const (
+	// Where we store the config file
+	CONFIGFILE     = "config.json"
+	OLD_CONFIGFILE = ".dockercfg"
+	INDEXSERVER    = "https://index.docker.io/v1/"
```

Comment:
I would too but I couldn't. :-(  I got into an annoying circular dependency.  If I move these back into registry then cliconfig imports registry and then registry needs to import cliconfig for the config stuff.  
Open to ideas on how to fix it though.